texastribune/donations
# Replace ds-toolbox git dependency with queso-ui npm dependency

#### What's this PR do?

Previously we were importing our style assets via a github repo dependency. Now, we're switching to pull from an npm version instead. No visual changes - just structural.

#### Why are we doing this? How does it help us?

It's a grand plan to simplify our design system iteration process. Splitting our design system repo into 2 repos helps us directly include our asset library (queso-ui) in /texastribune and include the task runner helpers (queso-tools) in tt-docker-base.

#### How should this be manually tested?

`yarn` - to swap out the dependencies
`make restart` - This should rebuild the CSS and SVG sprites with the built-in `yarn run dev`

Check out a few routes to make sure nothing looks broken. Generally if the icons aren't missing the footer doesn't look out of whack, everything is working as it should.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

Not really

#### What are the relevant tickets?

Off-sprint - just me constantly making our CSS more complicated. JK it'll all be worth it!

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [x] Checked for security implications? *( no new vulnerabilities added)*

#### queso roll-out plan

* [x] Convert _ds-toolbox_ compiling tasks to queso-tools
* [x] Convert _donations_ compiling tasks to queso-tools
* [x] Deprecate styles.js and icons.js and related dependencies in _ds-toolbox_
* [x] Restructure _ds-toolbox_ to form queso-ui package (leaving assets folder intact)
* [x] **Convert _donations_ asset grab to queso-ui** <= we are here (almost there!)
* [ ] Convert _texastribune_ asset grab to queso-ui
